lib/Service/RecordingService.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2022 Vitor Mattos <vitor@php.rio>
+ *
+ * @author Vitor Mattos <vitor@php.rio>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Service;
+
+use InvalidArgumentException;
+use OC\Files\Filesystem;
+use OC\User\NoUserException;
+use OCA\Talk\Config;
+use OCA\Talk\Exceptions\ParticipantNotFoundException;
+use OCA\Talk\Room;
+use OCP\Files\Folder;
+use OCP\Files\IMimeTypeDetector;
+use OCP\Files\IRootFolder;
+use OCP\Files\NotFoundException;
+use OCP\Files\NotPermittedException;
+
+class RecordingService {
+	private IMimeTypeDetector $mimeTypeDetector;
+	private ParticipantService $participantService;
+	private IRootFolder $rootFolder;
+	private Config $config;
+
+	public function __construct(
+		IMimeTypeDetector $mimeTypeDetector,
+		ParticipantService $participantService,
+		IRootFolder $rootFolder,
+		Config $config
+	) {
+		$this->mimeTypeDetector = $mimeTypeDetector;
+		$this->participantService = $participantService;
+		$this->rootFolder = $rootFolder;
+		$this->config = $config;
+	}
```

Comment:
We can now use constructor promotion (in new code)
https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion

```suggestion
	public function __construct(
		protected IMimeTypeDetector $mimeTypeDetector;
		protected ParticipantService $participantService;
		protected IRootFolder $rootFolder;
		protected Config $config;
	) {}
```